service/src/main/java/at/jku/isse/ecco/service/EccoService.java

```diff
@@ -1637,35 +1730,50 @@ public synchronized void addVariant(Configuration configuration, String name, Ec
      * Remove a new variant configuration
      *
      * @param configuration The configuration of the variant
-     * @param service       The service to remove a variant from the repository
      */
-    public synchronized void removeVariant(Configuration configuration, EccoService service) {
-        service.checkInitialized();
+    public synchronized void removeVariant(Configuration configuration) {
+        this.checkInitialized();
```

Comment:
In general `this.` can be removed if it is not needed, e.g., in the constructor it is usually needed since the parameter names match the field names. but here it doesn't look like it would be an issue